### PR TITLE
Closes #3272, #2438: Improve FirefoxAccountsAuthFeature to allow apps to decide how to open the FxA login.

### LINKED ISSUE #3272: Improve interface to create a FirefoxAccountsAuthFeature
Currently the [FirefoxAccountsAuthFeature](https://github.com/mozilla-mobile/android-components/blob/master/components/feature/accounts/src/main/java/mozilla/components/feature/accounts/FirefoxAccountsAuthFeature.kt) requires a TabUseCase. I would love to be able to provide my own method for dealing with the authURL to better control the auth flow.

see: https://github.com/mozilla-mobile/fenix/pull/3153

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.

### CODE REVIEW

File: components/feature/accounts/src/main/java/mozilla/components/feature/accounts/FirefoxAccountsAuthFeature.kt [hunk 1]
```diff
@@ -10,19 +10,23 @@ import kotlinx.coroutines.Dispatchers
...
-import mozilla.components.feature.tabs.TabsUseCases
 import mozilla.components.service.fxa.manager.FxaAccountManager
 import kotlin.coroutines.CoroutineContext
 
 /**
  * Ties together an account manager with a session manager/tabs implementation, facilitating an
  * authentication flow.
+ * @property accountManager .
+ * @property redirectUrl This is the url that will be reached at the final authentication state.
+ * @property coroutineContext The context where [onBeginAuthentication] will be executed,
```
Comment (grigoryk) [inline components/feature/accounts/src/main/java/mozilla/components/feature/accounts/FirefoxAccountsAuthFeature.kt]: `The context which will be used to execute network requests necessary to initiate authentication. Note that [onBeginAuthentication] will be executed using this context.`

File: components/feature/accounts/src/main/java/mozilla/components/feature/accounts/FirefoxAccountsAuthFeature.kt [hunk 2]
```diff
@@ -10,19 +10,23 @@ import kotlinx.coroutines.Dispatchers
...
+ * if you want an different context you are in charge of switching. The default context is [Dispatchers.IO].
+ * @property onBeginAuthentication A lambda function that receives the authentication url.
```
Comment (grigoryk) [inline components/feature/accounts/src/main/java/mozilla/components/feature/accounts/FirefoxAccountsAuthFeature.kt]: `A lambda function that receives the authentication url. Executed on [coroutineContext].`

Comment (Amejia481) [reply components/feature/accounts/src/main/java/mozilla/components/feature/accounts/FirefoxAccountsAuthFeature.kt]: Fixed!

File: components/feature/accounts/src/main/java/mozilla/components/feature/accounts/FirefoxAccountsAuthFeature.kt (hunk 1 above)
Comment (Amejia481) [reply components/feature/accounts/src/main/java/mozilla/components/feature/accounts/FirefoxAccountsAuthFeature.kt]: Fixed!

**Comment by grigoryk** [comment]
Ah! As I'm integrating this in Fenix, I'm hitting https://github.com/mozilla-mobile/fenix/pull/4476#issuecomment-517486303

Seems like we'll need to pass in the activity context.

**Comment by grigoryk** [comment]
Rewrote the patch in https://github.com/mozilla-mobile/android-components/pull/3995.

**Pull request closed without merging**